src/components/Drawer/Drawer.js

```diff
@@ -1,55 +1,112 @@
 
+const TRANSITION_DURATION = {
+    enter: 10,
+    exit: 300,
+};
+
 const { block, elem } = bem({
     name: 'Drawer',
     classnames: styles,
-    propsToMods: ['isOpen'],
+    propsToMods: ['isShownAndExpanded', 'isShownAndClosed', 'isExpanded'],
 });
 
 const Drawer = props => {
-    const { title, initialIsOpen, children, ...rest } = props;
-    const [isOpen, setIsOpen] = useState(initialIsOpen);
+    const {
+        title,
+        isShow,
+        isExpand: isExpandProps,
+        initialIsExpand,
+        onClick,
+        children,
+        ...rest
+    } = props;
+    const [isExpand, setIsExpand] = useState(isExpandProps || initialIsExpand);
+
+    useEffect(() => {
+        if (isExpandProps !== null) {
+            setIsExpand(isExpandProps);
+        }
+    }, [isExpandProps]);
+
+    function getStatus() {
+        const STATUS_LIST = {
+            IS_SHOWN_AND_EXPENDED: 'isShownAndExpanded',
+            IS_SHOWN_AND_CLOSED: 'isShownAndClosed',
+            EMPTY: '',
+        };
+
+        if (isShow) {
+            return isExpand ? STATUS_LIST.IS_SHOWN_AND_EXPENDED : STATUS_LIST.IS_SHOWN_AND_CLOSED;
+        }
+
+        return STATUS_LIST.EMPTY;
+    }
 
     function handleClick() {
-        setIsOpen(!isOpen);
+        if (onClick) {
+            onClick();
+        } else {
+            setIsExpand(!isExpand);
+        }
     }
 
-    return (
-        <div {...block({ isOpen })} {...rest}>
-            <header {...elem('head', props)}>
-                <h4 {...elem('headTitle', props)}>{title}</h4>
-                <div {...elem('headButtons', props)}>
-                    <button
-                        {...elem('extendButton', { isOpen })}
-                        type="button"
-                        onClick={handleClick}
-                    >
-                        <IoIosArrowUp {...elem('extendIcon', props)} />
-                    </button>
+    return ReactDOM.createPortal(
+        <Transition appear unmountOnExit in={isShow} timeout={TRANSITION_DURATION}>
+            {state => (
+                <div
+                    {...block({ [getStatus()]: state !== 'entering' })}
+                    aria-hidden={!isShow}
+                    {...rest}
+                >
+                    <header {...elem('head', props)}>
+                        <h4 {...elem('headTitle', props)}>{title}</h4>
+                        <div {...elem('headButtons', props)}>
+                            <button
+                                {...elem('expandButton', { isExpanded: isExpand })}
+                                type="button"
+                                onClick={handleClick}
+                            >
+                                <IoIosArrowUp {...elem('expandIcon', props)} />
+                            </button>
+                        </div>
+                    </header>
+                    <section {...elem('body', props)} aria-hidden={!(isShow && isExpand)}>
```

Comment:
shouldn't the body have some default padding around it? It looks a but funny now.

Reply:
I didn't think so. Always inner components have some padding/margin. If we add now some padding it will be very hard to align text inside inner components with Drawer title. But I agree with you, now it looks a little bit funny.

Reply:
I don't know the usecases this component solves, but if it is normally text (as in the example) that is shown inside the drawer, then adding padding makes sense. If it is usually something more complex, then there is no need to style it indeed.

Reply:
Yes, it is more complex :). In Search, we added notification and list of the queue. So padding here  will be redundant
![Screenshot 2019-09-23 at 15 08 14](https://user-images.githubusercontent.com/19436304/65428238-56e3e980-de14-11e9-9aa4-3566a1f50102.png)

Reply:
I see. Thanks.